### PR TITLE
feat(onboarding): replace the setup wizard with a chat-driven flow

### PR DESCRIPTION
**Pull request by geriamir**
Onboarding was a five-step MUI stepper — progress bar, panel, Next button. It worked, but it read like a form, and the steps where the server does slow work (a six-month scrape, credit card detection) showed a spinner with no sense of what was happening or why.

This replaces it with a conversation. **Same six steps, same endpoints, same server-side state machine — only the presentation changed.** There are no backend changes in this PR.

| Before | After |
|---|---|
| Stepper + panel + Next | Chat transcript + inline cards |
| Spinner during scrape | Typing indicator + live progress narration |
| Silent resume into step N | Conversation history, then the current question |

## Scripted, not generated

**No language model is involved at any point in this flow, and that is deliberate.** This flow handles bank credentials, so nothing in it is sent to a model. What ships is a deterministic state machine rendered as dialogue — a fixed script that reacts to onboarding status.

Per the design decision going in, **input is structured only**. There is no free-text box: every answer comes from a field card or a quick-reply button, so the flow cannot receive input it has no way to interpret.

## The transcript is derived, not stored

`buildScript(status)` is a pure function of `OnboardingStatus`. Two properties follow, and both are the reason it is built this way:

- **Reloading mid-setup rebuilds the same conversation**, because the state it is drawn from is the same state that decides what happens next. A stored transcript could drift out of step with actual progress; a derived one cannot.
- **Its only raw material is `/onboarding/status`, which carries no credentials.** The bubble echoing the user's answer *physically cannot* show a password, rather than relying on someone remembering to mask one. There is a test pinning this.

Message ids are semantic and stable (`greet`, `answer-checking`, `card:checking-account`) rather than positional, so a message keeps its identity as the conversation grows instead of renumbering on every step.

## Pacing

A first-time user has their opening lines revealed one at a time. Anyone resuming has already lived through that, so their history renders instantly — **the flow never replays what the user already did**. Respects `prefers-reduced-motion`. While the next move belongs to the server, a typing indicator stays up; silence in a chat reads as breakage.

## Three bugs fixed along the way

- **A partial status document crashed the page.** `buildScript` and the cards trusted every section of the payload to be present; one missing section threw and took down the whole page — the first page a new user sees. Now every section is read defensively. Caught by the e2e suite, now pinned by a unit test.
- **Both "skip cards" buttons rendered with no label** (`<Button ...></Button>`) in the old `CreditCardDetection`. Cypress clicked them by testid, so nothing ever caught it.
- **The bank and provider selects had labels not associated with their controls**, so screen readers never announced them.

## Testing

The Cypress contract was **preserved deliberately rather than rewritten**. All eleven testids and every text assertion were kept byte-identical to the old components, so the existing spec verifies the new UI against expectations written for the old one — a rewritten spec would have proven nothing about equivalence.

- **29/29 e2e tests pass** across all five specs, run locally against a real backend, all unmodified
- **228 frontend unit tests pass** (208 baseline + 20 new: script builder and chat shell)
- `tsc --noEmit`, `eslint --max-warnings 0`, and the production build are all clean
- Bundle is **2 kB smaller** — the stepper is gone

Docs: `ONBOARDING_TECHNICAL_ARCHITECTURE.md` updated to describe the chat components. References in `docs/archive/` were left alone as historical records.

### CODE REVIEW

File: frontend/src/components/onboarding/chat/OnboardingChat.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+const CARDS: Record<CardId, React.FC<CardProps>> = {
+  'checking-account': CheckingAccountCard,
+  'import-progress': ImportProgressCard,
+  'credit-card-choice': CreditCardChoiceCard,
+  'credit-card-form': CreditCardFormCard,
+  'matching-review': MatchingReviewCard,
+  complete: CompleteCard
+};
+
+const STEP_COUNT = 5;
```
[the remark this answers is not in the document]
Comment (geriamir) [reply frontend/src/components/onboarding/chat/OnboardingChat.tsx]: Good catch, and the bug is real — though the step count turned out not to be the cause.

`completedSteps` can't actually reach 6: `complete` is in the schema enum (`User.js:258`) but no route ever pushes it, so five was the right denominator.

The real problem is that the server writes the last two entries **before the user has acted on them**. In `onboardingEventHandlers.js:264-313`, when coverage comes back below 100%:

```js
const isFullCoverage = coverageAnalysis.coveragePercentage === 100;
const nextStep = isFullCoverage ? 'complete' : 'credit-card-matching';
...
'onboarding.currentStep': nextStep,
'onboarding.isComplete': isFullCoverage,
$addToSet: { 'onboarding.completedSteps': { $each: ['credit-card-setup', 'credit-card-matching'] } }
```

So on partial coverage all five steps are recorded while `isComplete` stays false and the user is parked on the matching review card — a full bar above a question they still have to answer. Partial coverage is an ordinary outcome, not an edge case.

Fixed by deriving progress from `currentStep`'s position rather than a count, since that's where the user actually is. Added a test for the partial-coverage case; reverting the fix fails it with exactly this symptom (received 100, expected < 100) and fails nothing else.

Also gave the header bar its own `onboarding-progress` testid — `progress-bar` belongs to the import card, so there was no way to assert on the header bar specifically.

51a3293

File: docs/ONBOARDING_TECHNICAL_ARCHITECTURE.md [hunk 2]
```diff
@@ -63,91 +67,102 @@ const OnboardingPage: React.FC = () => {
...
+
+**Input is structured only** - there is no free-text box. Every answer comes
+from a field card or a quick-reply button, so the flow cannot receive input it
+has no way to interpret.
 
-**Step Flow**:
+**Step Flow** (unchanged - the server still owns it):
 ```
 checking-account → transaction-import → credit-card-detection → credit-card-setup → complete
 ```
```
[the remark this answers is not in the document]
Comment (geriamir) [reply docs/ONBOARDING_TECHNICAL_ARCHITECTURE.md:91]: Correct — fixed in 51a3293.

`credit-card-matching` was missing from the step flow in two places, not one: the flow line here, and the numbered walkthrough in §3.1 Complete Onboarding Flow, which went straight from credit card setup to completion.

Both now include it, with a note that it's only surfaced when coverage comes back below 100% — at full coverage `onboardingEventHandlers` sets `currentStep` directly to `complete`, so the step is genuinely skipped rather than passed through.
